aces-nifi-processors/src/main/java/net/acesinc/nifi/processors/sockets/SocketIOProcessor.java

```diff
@@ -194,15 +195,25 @@ public void setupSockets(final ProcessContext context){
         //Try to create the socket is the URI is valid
         try {
             URI uri = new URI(url);
-            socket.set(createSocket(uri, opts));
 
             //If the scheme is HTTPS then create an SSLContext.
             if (uri.getScheme().equalsIgnoreCase(SCHEME_TYPE_HTTPS)) {
                 SSLContextService sslContextService = context.getProperty(SSL_CONTEXT_SERVICE).asControllerService(SSLContextService.class);
-                sslCon = sslContextService.createSSLContext(SSLContextService.ClientAuth.NONE);
-                opts.sslContext = sslCon;
+                sslCon = sslContextService.createContext();
+                X509TrustManager trustMan = sslContextService.createTrustManager();
+                OkHttpClient httpClient = new OkHttpClient().newBuilder().sslSocketFactory(sslCon.getSocketFactory(),
+                        trustMan).build();
+
                 opts.secure = true;
-                IO.setDefaultSSLContext(sslCon);
+                opts.callFactory = httpClient;
+                opts.webSocketFactory = httpClient;
+                IO.setDefaultOkHttpCallFactory(httpClient);
+                IO.setDefaultOkHttpWebSocketFactory(httpClient);
+
+                socket.set(createSocket(uri, opts));
```

Comment:
I'm counting like 3 new hoola-hoops to prepare a socket w/ an SSL context properly.